android/app/src/main/java/app/organicmaps/editor/LanguagesFragment.java

```diff
@@ -24,28 +25,43 @@ public interface Listener
     void onLanguageSelected(Language language);
   }
 
+  private Listener listener;
+
   @NonNull
   @Override
   protected LanguagesAdapter createAdapter()
   {
     Bundle args = getArguments();
-    Set<String> existingLanguages = new HashSet<>(args.getStringArrayList(EXISTING_LOCALIZED_NAMES));
+    Set<String> existingLanguages = args != null
+            ? new HashSet<>(args.getStringArrayList(EXISTING_LOCALIZED_NAMES))
+            : new HashSet<>();
 
     List<Language> languages = new ArrayList<>();
     for (Language lang : Editor.nativeGetSupportedLanguages())
     {
-      if (!existingLanguages.contains(lang.code))
-        languages.add(lang);
+      if (existingLanguages.contains(lang.code))
+        continue;
+      if (Editor.nativeIsServiceLanguage(lang.code))
```

Comment:
Can this check be moved into `nativeGetSupportedLanguages(bool returnServiceLanguages)` ?